### PR TITLE
Code refactoring and enforcing style guides #84

### PR DESCRIPTION
**Pull request by Lercerss**
Exhaustive refactoring across all components of the project for #84. Code styling was enforced according to [the Coding Style wiki page](https://github.com/Lercerss/SOEN341W18/wiki/Coding-Style). Although there are a lot of lines added/deleted, not that much has changed. Please review carefully!

**Refactoring done:**
  - Fixed indentation and spacing between symbols.
  - Removed unnecessary code/comments.
  - Added comments and docstrings where missing.
  - Renamed variables, functions, classes and files to proper casing.
  - Modified existing comments for clarity.

***

When reviewing, please take note of the following name changes:
### Templates
Previous          | New
--------------------|-------------------------
questionspage | posting_question
answerspage   | question_thread
UserProfile      | user_profile
EditUserProfile | edit_user_profile
question_display_page | question_index
edit                  | edit_post

### Urls
Previous          | New
--------------------|-------------------------
editAnswers    | edit_answers
QuestionIndex | question_index
editprofile        | edit_profile

### Models
Questions, Answers and Comments are now singular: Question, Answer, Comment
(This change will require you to rebuild your local index for whoosh)

### Static Files
`base.js` is now `questionthread.js`
`UserProfile.css` was deleted since it was not used.

### CODE REVIEW

File: SA3/qa_web/static/css/base.css [hunk 1]
```diff
@@ -1,69 +1,67 @@
-.good_answer{
-  color: green;
+.good-answer {
```
Comment (Belize22) [inline SA3/qa_web/static/css/base.css]: CSS properties like this need to be one line. Only use this formatting for CSS properties taking up multiple lines.

File: SA3/qa_web/views.py [hunk 2]
```diff
@@ -224,39 +276,45 @@ def vote(request):
...
         post.save()
-        return JsonResponse({'new_score' : post.score, 'id' : 'score_{}_{}'.format(post.id, post_type)})
+        return JsonResponse({'new_score': post.score, 'id': 'score_{}_{}'.format(post.id, post_type)})
     # Accessing url without using Post or unauthenticated
     return HttpResponseRedirect('/')
 
 
-# Home Page
 def homepage(request):
+    """Displays the website's home page.
```
Comment (Belize22) [inline SA3/qa_web/views.py]: This comment in particular seems to do nothing other than state the obvious (the function name makes this function pretty self explanatory).

File: SA3/qa_web/views.py [hunk 3]
```diff
@@ -266,187 +324,137 @@ def get_context_data(self, *args, **kwargs):
...
+                ellipsis_on_left = True
 
-            if len(left) != 0 and left[0] > 1:
-                first = True
+            if len(pages_on_left) != 0 and pages_on_left[0] > 1:
+                display_first_page = True
         else:
-            left = page_range[(page_number - 3) if (page_number - 3) > 0 else 0:page_number - 1]
-            right = page_range[page_number:page_number + 2]
+            pages_on_left = page_range[(page_number - 3) if (page_number - 3) > 0 else 0:page_number - 1]
```
[the remark this answers is not in the document]
Comment (Lercerss) [reply SA3/qa_web/views.py]: This is a ternary operator and not a `if cond: line`, therefore it is fine.

[the remark this answers is not in the document]
Comment (Belize22) [reply SA3/qa_web/views.py]: It checks out. Apologies for the false alarm.

File: SA3/qa_web/static/css/base.css (hunk 1 above)
Comment (Lercerss) [reply SA3/qa_web/static/css/base.css]: In the [Google HTML/CSS Style Guide](https://google.github.io/styleguide/htmlcssguide.html#CSS), this is the preferred way of applying properties, even if they do not take mutliple lines.

**Comment by Lercerss** [comment]
Thanks @Belize22 for your review.
  - testModal.html does not exist on master
  - Indents are intentionally 4 spaces in html files to match all other files in the project.
  - I will correct lines that exceed the maximum where possible. (python & html)
  - Title and meta tags are found in the base.html, therefore all files have them.

**Comment by Belize22** [comment]
For the HTML indenting, you may wanna specify it is 4 spaces in the wiki page for coding style as a way to make it supercede the 2 indent rule for HTML and to avoid potential confusion when one looks at both the code and the styling guidelines

**Comment by Lercerss** [comment]
It is already the case: 

> For better consistency with other languages used in the project, files are indented with 4 spaces.

Thanks for being thorough.

File: SA3/qa_web/models.py [hunk 4]
```diff
@@ -1,90 +1,96 @@
...
+                                 on_delete=models.CASCADE)
+    correct_answer = models.BooleanField(default=False)
+    voters = models.ManyToManyField(
+        settings.AUTH_USER_MODEL, related_name='a_voters', through='Vote')
+
+
+class Comment(Post):
+    """A Comment is a lower-level response to either an Answer or Question.
+    It is the second level of discussion as it might not directly relate to
+    the Question's matter.
```
Comment (alvyn279) [inline SA3/qa_web/models.py]: Nice comments for every single class in `models.py`.

File: SA3/qa_web/templates/qa_web/includes/editor_style.html [hunk 5]
```diff
@@ -1,28 +1,28 @@
 {% load static %}
```
Comment (alvyn279) [inline SA3/qa_web/templates/qa_web/includes/editor_style.html]: Still weirded out by the existence of this file, but it is necessary for the display of buttons on the MD Editor.

File: SA3/qa_web/views.py [hunk 6]
```diff
@@ -1,78 +1,93 @@
...
             user.email = form.cleaned_data.get('email')
             user.birthday = form.cleaned_data.get('birthday')
-            user.motherland = request.POST['motherland']
-            user.school = request.POST['school']
-            user.major = request.POST['major']
-            user.city = request.POST['city']
+            user.motherland = form.cleaned_data.get('motherland')
+            user.school = form.cleaned_data.get('school')
+            user.major = form.cleaned_data.get('major')
+            user.city = form.cleaned_data.get('city')
```
Comment (alvyn279) [inline SA3/qa_web/views.py]: Great to finally have `form.cleaned_data` as we have discussed many times on wanting to unify this.

**Pull request merged**